app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class SessionsController < ApplicationController
+  layout 'auth', only: %i[new create]
+
+  before_action :redirect_to_home, only: %i[new create]
```

Comment:
I think the usage of the before action `redirect_to_home` here isn't clear enough & may cause confusion. Reading this line of code, one can question why we need to redirect to the home page for these actions? It turns out we need to do that if the user already logged in only.
So how about removing the condition checking in that method
```ruby
  def redirect_to_home
    redirect_to root_path
  end
```

and bring that condition here as:
```ruby
before_action :redirect_to_home, only: %i[new create], if: current_user
```
what do you think @andyduong1920 @gutakk ?

Reply:
Good catch, I agree with you that this is quite confusing.

how about this?
```ruby
before_action(only: %i[new create]) { redirect_to root_path if current_user }
```
if we do this way then we don't need `redirect_to_home` method which just only called `redirect_to root_path` anymore.

Reply:
I prefer to keep the `redirect_to_home` method since we can reuse it elsewhere.
beside that, that syntax looks complicated
```
before_action(only: %i[new create]) { redirect_to root_path if current_user }
```
than extracting it to a separted method, I think. Do you have any inputs @andyduong1920 ?

Reply:
Agree with syntax looks complicated, will make the change.

Reply:
Fixed in 714daa633f24a12a6b46ac5ba24b5367c4c7e195